pkg/node/manager/queue.go

```diff
@@ -24,15 +24,15 @@ func (q *queue[T]) isEmpty() bool {
 	return len(q.items) == 0
 }
 
-func (q *queue[T]) pop() *T {
-	if q.isEmpty() {
```

Comment:
nit: this is a slight semantic change - previously, when the queue was empty `pop` could proceed in the face of other readers, now it cannot. Can't think of why this would break something, though.